app/routes/todos.tsx

```diff
@@ -44,94 +58,201 @@ export function meta() {
   ];
 }
 
-export default function Todos({ loaderData }: Route.ComponentProps) {
-  const { todos } = loaderData;
+interface KanbanContextType {
+  draggedId: string | null;
+  setDraggedId: (id: string | null) => void;
+}
 
+const KanbanContext = createContext<KanbanContextType | null>(null);
+
+function KanbanProvider({ children }: { children: ReactNode }) {
+  const [draggedId, setDraggedId] = useState<string | null>(null);
   return (
-    <div className="min-h-screen bg-gray-50 py-8">
-      <div className="max-w-md mx-auto bg-white rounded-lg shadow-md p-6">
-        <div className="mb-6">
-          <h1 className="text-2xl font-bold text-gray-800">Todos</h1>
-        </div>
-        
-        {/* Add todo form */}
-        <Form method="post" className="mb-6">
-          <input type="hidden" name="intent" value="add" />
-          <div className="flex gap-2">
-            <input
-              type="text"
-              name="text"
-              placeholder="What needs to be done?"
-              className="flex-1 px-3 py-2 border border-gray-300 rounded-md focus:outline-none focus:ring-2 focus:ring-blue-500"
-              required
-            />
+    <KanbanContext.Provider value={{ draggedId, setDraggedId }}>
+      {children}
+    </KanbanContext.Provider>
+  );
+}
+
+function useKanban() {
+  const context = useContext(KanbanContext);
+  if (!context) throw new Error("useKanban must be used within KanbanProvider");
+  return context;
+}
+
+function FormBridge() {
+  const submit = useSubmit();
+  return {
+    submit: (data: Record<string, any>) => {
+      const formData = new FormData();
+      Object.entries(data).forEach(([key, value]) => {
+        formData.append(key, String(value));
+      });
+      submit(formData, { method: "post" });
+    }
+  };
+}
+
+function KanbanCard({ todo }: { todo: Todo }) {
+  const { setDraggedId } = useKanban();
+  const formBridge = FormBridge();
+
+  const handleDragStart = (e: React.DragEvent) => {
+    setDraggedId(todo.id);
+    e.dataTransfer.setData('text/plain', todo.id);
+  };
+
+  const handleDragEnd = () => {
+    setDraggedId(null);
+  };
+
+  return (
+    <div
+      draggable
+      onDragStart={handleDragStart}
+      onDragEnd={handleDragEnd}
```

Comment:
issue (reusability): export new hook (better in new file) for drag and drop functionality and extend usability on board for the columns and rows, and other components in future.
```suggestion
  export function useDraggable(id: string, onDragStart?: () => void, onDragEnd?: () => void) {
  const handleDragStart = useCallback((e: React.DragEvent) => {
    e.dataTransfer.setData('text/plain', id);
    onDragStart?.();
  }, [id, onDragStart]);

  const handleDragEnd = useCallback(() => {
    onDragEnd?.();
  }, [onDragEnd]);

  return { draggable: true, onDragStart: handleDragStart, onDragEnd: handleDragEnd };
}
  
  const { draggable, onDragStart, onDragEnd } = useDraggable(
    todo.id,
    () => setDraggedId(todo.id),
    () => setDraggedId(null)
  );

  return (
    <div
      draggable
      {...{ draggable, onDragStart, onDragEnd }}
```